data-import/desktop-csv-import/desktop-csv-import.adoc

```diff
@@ -0,0 +1,420 @@
+The ones we will use for this exercise are as follows:
+
+* *toInteger():* converts a value to an integer.
+* *toFloat():* converts a value to a float (in this case, for monetary amounts).
+* *datetime():* converts a value to a datetime.
+
+We will look at the values in each CSV file to determine what needs to be converted.
+
+.Products.csv
+The values in the products.csv files are for product ID, product name, and unit cost.
+Product ID looks to be an integer value that increases with each row, so we can convert this to an integer using the `toInteger()` function in Cypher.
+Product name can remain a string because it consists of characters.
+The final column is the product unit cost.
+Though the sample values from our inspection above are all whole numbers, we know that monetary amounts can have decimal place values, as well, so we will convert these values to floats using the `toFloat()` function.
+
+We can see the Cypher to handle all of these conversions below; however, we are still not loading the values into Neo4j yet.
+We are just viewing the CSV files with converted values.
+
+[source, cypher]
+----
+LOAD CSV FROM 'file:///desktop-csv-import/products.csv' AS row
+RETURN toInteger(row[0]), row[1], toFloat(row[2])
+LIMIT 3
+----
+
+image::{img}desktop_import_products_convert.jpg[link="{img}desktop_import_products_convert.jpg",role="popup-link"]
+
+Note that we are using collection positions (row[0], row[1], row[2]) to refer to the columns in the row.
+In a file that has no headers, this is the way to reference values in each position.
+When we look at the next two files that have headers, you will see that we can use the column name to reference the value.
+
+.Orders.csv
+The values in the orders.csv (per the column names) are for orderID, orderDate, and shipCountry.
+Again, we can evaluate the values and determine any conversions to apply.
+
+OrderID looks like it is an integer, so we can convert that using the `toInteger()` function.
+The orderDate column is certainly in a date format and will require us to format it using the `datetime()` function.
+Finally, the shipCountry values are characters, so we can leave that as a string.
+
+[NOTE]
+--
+If you are using a version of Neo4j prior to 3.4, the `datetime()` function will not be available.
+That function was released in 3.4 and is supported from that version forward.
+--
+
+Just as we did with the last CSV file, let us look at the results of these conversions without importing the data.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/orders.csv' AS row
+RETURN toInteger(row.orderID), datetime(replace(row.orderDate,' ','T')), row.shipCountry
+LIMIT 5
+----
+
+image::{img}desktop_import_orders_convert.jpg[link="{img}desktop_import_orders_convert.jpg",role="popup-link"]
+
+There was one tricky thing with this CSV in the `orderDate` column.
+Neo4j's datetime format will accept many formats, but the time value must start with a `T`.
```

Comment:
that's actually ISO 8601 date format:
https://en.wikipedia.org/wiki/ISO_8601

Reply:
Ah, thank you! I'll reword that sentence.